sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/RebaseDateTime.scala

```diff
@@ -363,33 +371,46 @@ object RebaseDateTime {
   }
 
   /**
-   * An optimized version of [[rebaseGregorianToJulianMicros(ZoneId, Long)]]. This method leverages
-   * the pre-calculated rebasing maps to save calculation. If the rebasing map doesn't contain
-   * information about the current JVM system time zone or `micros` is related to Before Common Era,
-   * the function falls back to the regular unoptimized version.
-   *
-   * Note: The function assumes that the input micros was derived from a local timestamp
-   *       at the default system JVM time zone in Proleptic Gregorian calendar.
+   * An optimized version of [[rebaseGregorianToJulianMicros(TimeZone, Long)]]. This method
+   * leverages the pre-calculated rebasing maps to save calculation. If the rebasing map doesn't
+   * contain information about the given time zone `timeZoneId` or `micros` is related to Before
+   * Common Era, the function falls back to the regular unoptimized version.
    *
+   * @param timeZoneId A string identifier of a time zone.
    * @param micros The number of microseconds since the epoch '1970-01-01T00:00:00Z'
    *               in Proleptic Gregorian calendar. It can be negative.
    * @return The rebased microseconds since the epoch in Julian calendar.
    */
-  def rebaseGregorianToJulianMicros(micros: Long): Long = {
+  def rebaseGregorianToJulianMicros(timeZoneId: String, micros: Long): Long = {
     if (micros >= lastSwitchGregorianTs) {
       micros
     } else {
-      val timeZone = TimeZone.getDefault
-      val tzId = timeZone.getID
-      val rebaseRecord = gregJulianRebaseMap.getOrNull(tzId)
+      val rebaseRecord = gregJulianRebaseMap.getOrNull(timeZoneId)
       if (rebaseRecord == null || micros < rebaseRecord.switches(0)) {
-        rebaseGregorianToJulianMicros(timeZone, micros)
+        rebaseGregorianToJulianMicros(TimeZone.getTimeZone(timeZoneId), micros)
       } else {
         rebaseMicros(rebaseRecord, micros)
       }
     }
   }
 
+  /**
+   * An optimized version of [[rebaseGregorianToJulianMicros(TimeZone, Long)]]. This method
+   * leverages the pre-calculated rebasing maps to save calculation. If the rebasing map doesn't
+   * contain information about the current JVM system time zone or `micros` is related to Before
+   * Common Era, the function falls back to the regular unoptimized version.
+   *
+   * Note: The function assumes that the input micros was derived from a local timestamp
+   *       at the default system JVM time zone in Proleptic Gregorian calendar.
+   *
+   * @param micros The number of microseconds since the epoch '1970-01-01T00:00:00Z'
+   *               in Proleptic Gregorian calendar. It can be negative.
+   * @return The rebased microseconds since the epoch in Julian calendar.
+   */
+  def rebaseGregorianToJulianMicros(micros: Long): Long = {
```

Comment:
is it used in test only?

Reply:
Not only, see
https://github.com/apache/spark/blob/f7dabd8e57fc4edf057b9219d6d9382bc6adf749/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala#L328